syntax = "proto3";

option go_package = "digitalSignage/api";
package DeviceRegistration;

service DeviceRegistration {
    rpc RegisterDevice(DeviceRegistrationRequest) returns (DeviceRegistrationResponse) {}
}

enum RegistrationStatus {
    REG_SUCCESSFUL = 0;
    REG_UNSUCCESSFUL = 1;
    REG_ERROR = 2;
  }

message DeviceRegistrationRequest{
    string device_mac_add=1;
    string device_unique_id=2;
    string fw_version=3;
    string os_version=4;
}

message DeviceRegistrationResponse{
    RegistrationStatus status=1;
    string server_url=2;
    string unique_system_id=3;
    string error_msg=4;
}